staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go

```diff
@@ -289,6 +294,98 @@ type Object interface {
 	Merged() (runtime.Object, error)
 
 	Name() string
+	GroupVersionKind() schema.GroupVersionKind
+}
+
+// mask conceals any secret values and returns the live/merged versions
+// of the object.
+//
+// mask only operates on V1Secret objects; any other objects will
+// be returned as-is.
+//
+// All secret values in the objects will be masked with a fixed-length
+// asterisk mask. If two values are different, an additional suffix will
+// be added so they can be diffed.
+func mask(obj Object) (live, merged runtime.Object, err error) {
```

Comment:
Still a lot of things are duplicated here, is there any way you can reduce the duplication further?

The fact that you return so many values here smells like something is wrong.